lib/reconstruct.ts

```diff
@@ -0,0 +1,75 @@
+import { types, Root, Token, Tokens, Group } from './types';
+import * as sets from './sets';
+
+const simplifications: [sets.SetFunc, string][] = [
+  [sets.words, '\\w'],
+  [sets.notWords, '\\W'],
+  [sets.ints, '\\d'],
+  [sets.notInts, '\\D'],
+  [sets.whitespace, '\\s'],
+  [sets.notWhitespace, '\\S'],
+  [sets.anyChar, '.']
+];
+
+const reduceStack = (stack: Token[]): string => stack.map(partialConstruct).join('');
+
+const createAlternate = (token: Root | Group): string => {
+  if ('options' in token) {
+    return token.options.map(reduceStack).join('|');
+  } else if ('stack' in token) {
+    return reduceStack(token.stack);
+  } else {
+    throw new Error(`options or stack must be Root or Group token`);
+  }
+};
+
+export const reconstruct = (regexpToken: Root): string => partialConstruct(regexpToken);
+
+export const partialConstruct = (token: Tokens): string => {
```

Comment:
what do you think of having only `reconstruct` and have it function like `partialConstruct`? imo there would be less confusion on when to use which

Reply:
Seems reasonable to me